improve-digital-htb.js

```diff
@@ -249,73 +265,77 @@ function ImproveDigitalHtb(configs) {
             headerStatsInfo[htSlotId] = {};
             headerStatsInfo[htSlotId][curReturnParcel.requestId] = [curReturnParcel.xSlotName];
 
-            var curBid;
-
-            for (var i = 0; i < bids.length; i++) {
-
-                /**
-                 * This section maps internal returnParcels and demand returned from the bid request.
-                 * In order to match them correctly, they must be matched via some criteria. This
-                 * is usually some sort of placements or inventory codes. Please replace the someCriteria
-                 * key to a key that represents the placement in the configuration and in the bid responses.
-                 */
-
-                /* ----------- Fill this out to find a matching bid for the current parcel ------------- */
-                if (curReturnParcel.xSlotRef.someCriteria === bids[i].someCriteria) {
-                    curBid = bids[i];
-                    bids.splice(i, 1);
-                    break;
-                }
-            }
-
-            /* No matching bid found so its a pass */
-            if (!curBid) {
-                if (__profile.enabledAnalytics.requestTime) {
-                    __baseClass._emitStatsEvent(sessionId, 'hs_slot_pass', headerStatsInfo);
-                }
-                curReturnParcel.pass = true;
-                continue;
+            var curBid = null;
+      
+            if (adResponse && adResponse.bid && adResponse.bid[0]) {
+              curBid = adResponse.bid[0];
+            } else {
+              if (__profile.enabledAnalytics.requestTime) {
+                __baseClass._emitStatsEvent(sessionId, 'hs_slot_pass', headerStatsInfo);
+              }
+              curReturnParcel.pass = true;
+              continue;
             }
-
+            
             /* ---------- Fill the bid variables with data from the bid response here. ------------*/
 
             /* Using the above variable, curBid, extract various information about the bid and assign it to
              * these local variables */
 
             /* the bid price for the given slot */
-            var bidPrice = curBid.price;
+            var bidPrice;
+            var bidIsPass;
+            if (typeof curBid.price === 'undefined' || curBid.price === 0) {
+                bidPrice = 0;
+                bidIsPass = true;
+            } else {
+                bidPrice = curBid.price * 100;
```

Comment:
This is handled by the adapter using the `bidUnitInCents`. It looks like the price returned by your ad server is in dollar, so please change the bidUnitInCents value to 100 and keep the bidPrice as the actual price returned from your end point.

Reply:
Fixed